src/nspawn/nspawn-mount.c

```diff
@@ -1349,3 +1349,115 @@ int setup_volatile(
         (void) rmdir(template);
         return r;
 }
+
+int pivot_root_parse(char **pivot_root_new, char **pivot_root_old, const char *s) {
+        _cleanup_free_ char *root_new = NULL, *root_old = NULL;
+        const char *p = s;
+        int r;
+
+        assert(pivot_root_new);
+        assert(pivot_root_old);
+
+        r = extract_first_word(&p, &root_new, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return -EINVAL;
+
+        if (isempty(p)) {
+                root_old = NULL;
+        } else {
+                root_old = strdup(p);
+                if (!root_old)
+                        return -ENOMEM;
+        }
+
+        if (!path_is_absolute(root_new))
+                return -EINVAL;
+        if (root_old && !path_is_absolute(root_old))
+                return -EINVAL;
+
+        free_and_replace(*pivot_root_new, root_new);
+        free_and_replace(*pivot_root_old, root_old);
```

Comment:
hmpf. Not a big fan of functions that have return parameters that have to be valid on input... But OK. Could you add a comment though to the top of the functoin that makes clear that this function expects the output parameters to be initialized?